exchanges/bybit/bybit_websocket.go

```diff
@@ -0,0 +1,483 @@
+
+	for i := range channelsToUnsubscribe {
+		var unSub WsReq
+		unSub.Event = cancel
+		unSub.Topic = channelsToUnsubscribe[i].Channel
+
+		formattedPair, err := by.FormatExchangeCurrency(channelsToUnsubscribe[i].Currency, asset.Spot)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		unSub.Parameters = WsParams{
+			Symbol: formattedPair.String(),
+		}
+		err = by.Websocket.Conn.SendJSONMessage(unSub)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		by.Websocket.RemoveSuccessfulUnsubscriptions(channelsToUnsubscribe[i])
+	}
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+// wsReadData receives and passes on websocket messages for processing
+func (by *Bybit) wsReadData(ws stream.Connection) {
+	defer by.Websocket.Wg.Done()
+	for {
+		resp := ws.ReadMessage()
+		if resp.Raw == nil {
+			return
+		}
+		comms <- resp
+	}
+}
+
+// GenerateDefaultSubscriptions generates default subscription
+func (by *Bybit) GenerateDefaultSubscriptions() ([]stream.ChannelSubscription, error) {
+	var subscriptions []stream.ChannelSubscription
+	var channels = []string{wsTicker, wsTrades, wsOrderbook, wsKlines}
+	pairs, err := by.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		return nil, err
+	}
+	for z := range pairs {
+		for x := range channels {
+			subscriptions = append(subscriptions,
+				stream.ChannelSubscription{
+					Channel:  channels[x],
+					Currency: pairs[z],
+					Asset:    asset.Spot,
+				})
+		}
+	}
+	return subscriptions, nil
+}
+
+func stringToOrderStatus(status string) (order.Status, error) {
+	switch status {
+	case "NEW":
+		return order.New, nil
+	case "CANCELED":
+		return order.Cancelled, nil
+	case "REJECTED":
+		return order.Rejected, nil
+	case "TRADE":
+		return order.PartiallyFilled, nil
+	case "EXPIRED":
+		return order.Expired, nil
+	default:
+		return order.UnknownStatus, errors.New(status + " not recognised as order status")
+	}
+}
+
+// WsDataHandler handles data from wsReadData
+func (by *Bybit) WsDataHandler() {
+	defer by.Websocket.Wg.Done()
+	for {
+		select {
+		case <-by.Websocket.ShutdownC:
+			return
+		case resp := <-comms:
+			err := by.wsHandleData(resp.Raw)
+			if err != nil {
+				by.Websocket.DataHandler <- err
+			}
+		}
+	}
+}
+
+func (by *Bybit) wsHandleData(respRaw []byte) error {
+	var multiStreamData map[string]interface{}
+	err := json.Unmarshal(respRaw, &multiStreamData)
+	if err != nil {
+		return err
+	}
+
+	if method, ok := multiStreamData["event"].(string); ok {
+		if strings.EqualFold(method, sub) {
+			return nil
+		}
+		if strings.EqualFold(method, cancel) {
+			return nil
+		}
+	}
+
+	if e, ok := multiStreamData["e"].(string); ok {
+		switch e {
+		case wsAccountInfoStr:
+			var data wsAccountInfo
+			err := json.Unmarshal(respRaw, &data)
+			if err != nil {
+				return fmt.Errorf("%v - Could not convert to outboundAccountInfo structure %s",
+					by.Name,
+					err)
+			}
+			by.Websocket.DataHandler <- data
+			return nil
+		case wsOrderStr:
```

Comment:
Using the [provided JSON data](https://bybit-exchange.github.io/docs/spot/#t-executionreport) and the following test:
```go

func TestWSOrderExecution(t *testing.T) {
	t.Parallel()
	pressXToJSON := []byte(`[
  {
    "e": "executionReport",      
    "E": "1499405658658",            
    "s": "ETHBTC",                 
    "c": "1000087761",               
    "S": "BUY",                    
    "o": "LIMIT",                  
    "f": "GTC",                    
    "q": "1.00000000",             
    "p": "0.10264410",             
    "X": "NEW",                    
    "i": "4293153",     
    "M": "0",             
    "l": "0.00000000",             
    "z": "0.00000000",             
    "L": "0.00000000",             
    "n": "0",                      
    "N": "BTC",                     
    "u": true,                     
    "w": true,                     
    "m": false,                    
    "O": "1499405658657",            
    "Z": "473.199",
    "A": "0",
    "C": false,
    "v": "0"              
  }
]
`)
	err := b.wsHandleData(pressXToJSON)
	if err != nil {
		t.Fatal(err)
	}
}

```

It does not look like this function works as intended

Reply:
You are correct @gloriousCode 
I have made the necessary changes.